src/state_machine.zig

```diff
@@ -1115,8 +1100,9 @@ pub fn StateMachineType(
         ) usize {
             _ = client;
             assert(op != 0);
-            assert(input_valid(operation, input));
+            assert(self.input_valid(operation, input));
             assert(timestamp > self.commit_timestamp or global_constants.aof_recovery);
+            assert(input.len <= self.batch_size_limit);
```

Comment:
Assert postcondition, that result <= self.batch_size_limit

Reply:
The result is allowed to exceed the `batch_size_limit` -- the batch limit is only for requests, not replies.

Reply:
Hm, then why do we compute it as 


```
                    const batch_limit = @min(
                        @divFloor(self.batch_size_limit, event_size),
                        @divFloor(self.batch_size_limit, result_size),
                    );
```

in `operation_batch_max`?

Reply:
Actually now I'm confused.. I think this behavior changed when I wasn't paying attention :sweat_smile: